### PR TITLE
js: use Promise for WebRTC operations

### PR DESCRIPTION
**Pull request by fippo**
also get rid of some RTC* constructors

(most important question: did i get the indent right?)

### CODE REVIEW

**Comment by lminiero** [comment]
Thanks, this looks like a huge improvement and update! I'm still out of the office, but I'll review as soon as I get back in a couple of days :+1:

File: html/janus.js [hunk 1]
```diff
@@ -1664,29 +1664,28 @@ function Janus(gatewayCallbacks) {
...
-							config.candidates = [];
 						}
-						// Create the answer now
-						createAnswer(handleId, media, callbacks);
-					}, callbacks.error);
+						config.candidates = [];
+					}
+					// Create the answer now
+					createAnswer(handleId, media, callbacks);
+				}, callbacks.error);
```
[the remark this answers is not in the document]
Comment (fippo) [reply html/janus.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then
-- the second argument to Promise is what should happen in the case of an error.
The semantic difference to adding a .catch is that the .catch will be triggered for anything in the .then as well whereas having callbacks.error as the second argument to the .then will only trigger for 'raw' WebRTC errors.

Does that make sense?

Now go back to :sun_with_face: :beach_umbrella:

[the remark this answers is not in the document]
Comment (lminiero) [reply html/janus.js]: Whoops, sorry about the dumb queston then! :sweat_smile:

[the remark this answers is not in the document]
Comment (fippo) [reply html/janus.js]: that is what I said when I first learned about this syntax :-)

**Comment by lminiero** [comment]
Just tested and it seems to work fine, thanks again! I don't see any issue with indentation: actually, I think you fixed it in a couple of parts... merging! :+1:

**Pull request merged**